### PR TITLE
HIVE-26196: Integrate Sonar analysis for the master branch and PRs

### PR DESCRIPTION
**Pull request by asolimando**
### What changes were proposed in this pull request?

Adding Sonar analysis (no test coverage) for the master branch and PRs only at the moment, to track code quality metrics over time.

### Why are the changes needed?

By running Sonar analysis regularly, we can catch critical bugs and issues more easily.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Sonar has been run locally and via CI (https://sonarcloud.io/summary/overall?id=apache_hive).

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -245,6 +245,21 @@ fi
         buildHive("install -Dtest=noMatches")
       }
       checkPrHead()
+      stage('Sonar') {
+        if(env.CHANGE_BRANCH == 'master-sonar_analysis') {
```
Comment (kgyrtkirk) [inline Jenkinsfile:249]: I think this should be changed to master - or not?

File: Jenkinsfile [hunk 2]
```diff
@@ -245,6 +245,21 @@ fi
...
+          withCredentials([string(credentialsId: 'sonar', variable: 'SONAR_TOKEN')]) {
```
Comment (kgyrtkirk) [inline Jenkinsfile:250]: nice :)

**Comment by asolimando** [comment]
Dear bot, please keep the PR open!

File: Jenkinsfile [hunk 3]
```diff
@@ -304,6 +304,53 @@ tar -xzf packaging/target/apache-hive-*-nightly-*-src.tar.gz
...
+                       -Dsonar.projectKey=apache_hive \
+                       -Dsonar.host.url=https://sonarcloud.io \
+                       -Dsonar.pullrequest.github.repository=apache/hive \
+                       -Dsonar.pullrequest.key=${CHANGE_ID} \
+                       -Dsonar.pullrequest.branch=${CHANGE_BRANCH} \
+                       -Dsonar.pullrequest.base=${CHANGE_TARGET} \
+                       -Dsonar.pullrequest.provider=GitHub \
+                       -DskipTests -Dit.skipTests -Dmaven.javadoc.skip
+                      """
+                 }
```
Comment (zabetak) [inline Jenkinsfile:347]: There are many similarities in this code snippet. You can probably refactor the moving part (different properties) in a variable (using the `if` branches) and call afterwards mvn.

**Review by zabetak — COMMENTED** [review]
Overall, LGTM. I left some minor comments.

File: Jenkinsfile (hunk 3 above)
Comment (asolimando) [reply Jenkinsfile]: I have unified the common pieces and created a parameterized function to run sonar analysis, current run is still on-going but the sonar task has already successfully completed

**Comment by asolimando** [comment]
The test failures are due to flakyness and not related at all to the present PR, I am not going to run tests again to avoid wasting resources, I hope that's OK with you @zabetak

**Review by zabetak — APPROVED** [review]
LGTM, will merge soon!

**Pull request closed without merging**